### ISSUE TITLE
Reading or writing to a database not configured in WAL mode will perform excessive I/O and reduce the health of storage devices. This is known to cause excessive I/O for your users.

### ISSUE THREAD

**Issue by amruth-movano** [aws-amplify/amplify-swift#3890]
### Describe the bug

We are getting high disk usage

### Steps To Reproduce

```swift
Configure amplify
Launch app
Use app
```

### Expected behavior

It should not give a warning for high disk usage.

### Amplify Framework Version

2.39.0

### Amplify Categories

DataStore

### Dependency manager

Swift PM

### Swift version

5.9.2

### CLI version

12.11.1

### Xcode version

15.2

### Relevant log output

```shell
-
```

### Is this a regression?

Yes

### Regression additional context

<img width="1445" alt="Screenshot 2024-10-01 at 11 36 46 PM" src="https://github.com/user-attachments/assets/8bacea80-a2a4-4f07-86a1-36028b603179">
<img width="1512" alt="Screenshot 2024-10-01 at 11 37 02 PM" src="https://github.com/user-attachments/assets/1f1246c9-cd54-42d5-b93e-cf3c30dea46f">

### Platforms

iOS

### OS Version

iOS 18.0

### Device

iPhone 12

**Comment by amruth-movano** [comment]
We are using 14 models.
On launch, we do amplify.start synchronization and getting this error

**Comment by amruth-movano** [comment]
@5d  yes its begin to happen on app launch and memory increasing in MBs. cpu usage also increasing to 100-200 %

**Comment by amruth-movano** [comment]
@5d is there any update on this?

**Comment by thisisabhash** [comment]
Hello, Would you be able to post more context about the issue? 
- How many models and number of records per model are being synced when you call `Amplify.Datastore.start()` ?
- Have you considered selective sync for your use case? https://docs.amplify.aws/gen1/swift/build-a-backend/more-features/datastore/sync-to-cloud/#selectively-syncing-a-subset-of-your-data
- Would you be able to share a sample schema and code snippets, which can help us reproduce the issue locally?

**Comment by vincetran** [comment]
@amruth-movano Can you provide code snippets to how you're calling this so that we can try and reproduce the issue?

**Comment by harsh62** [comment]
Closing stale issue. Please reopen if you are still facing this issue.
